src/main/webapp/components/workspace/workspace.js

```diff
@@ -94,22 +96,13 @@ export default () => {
   const [queries, setQueries] = useState()
   const { results, status, onSearch, onCancel, onClear } = useQueryExecutor()
 
-  const [queuedSearch, setQueuedSearch] = useState('')
-
-  React.useEffect(
-    () => {
-      if (queuedSearch === '') {
-        return
-      }
-      const index = queries.findIndex(query => query.id === queuedSearch)
-      if (index !== -1) {
-        const search = queryToSearch(queries[index])
-        onSearch(search)
-        setQueuedSearch('')
-      }
-    },
-    [queries, queuedSearch]
-  )
+  const saveWorkspace = useSaveWorkspace()
+  const createQuery = useCreateQuery(query => {
+    setQueries(merge([query], queries))
+    setCurrentQuery(query.id)
+    saveWorkspace({ queries: merge([query], queries) })
```

Comment:
:pencil2: could also use es6 syntax here
```suggestion
    saveWorkspace({ queries: [query, ...queries] })
```